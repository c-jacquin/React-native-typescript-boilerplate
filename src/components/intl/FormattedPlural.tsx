import React, { StatelessComponent } from 'react'
import { FormattedPlural } from 'react-intl'
import { Text } from 'react-native'

import { IFormattedPlural } from './types'

const NativeFormattedPlural: StatelessComponent<IFormattedPlural> = props => {
    return (
        <FormattedPlural {...props}>
            {(localized: string) =>
                <Text style={props.textStyle}>
                    {localized}
                </Text>}
        </FormattedPlural>
    )
}

export default NativeFormattedPlural
