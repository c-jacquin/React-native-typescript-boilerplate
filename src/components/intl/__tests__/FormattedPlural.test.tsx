import * as React from 'react'
import { Text } from 'react-native'
import * as renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'

import FormattedPlural from '../FormattedPlural'

describe('FormattedPlural', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <IntlProvider locale="en">
                    <FormattedPlural
                        style={'ordinal'}
                        value={'foo'}
                        other={<Text>Foo</Text>}
                    />
                </IntlProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
