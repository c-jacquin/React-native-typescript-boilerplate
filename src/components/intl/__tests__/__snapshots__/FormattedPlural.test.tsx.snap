// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedPlural should renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Foo
  </Text>
</Text>
`;
